api/src/swagger/config.yaml

```diff
@@ -48,4 +50,266 @@ paths:
         "200":
           description: "Successful operation"
         "404":
-          description: "Invalid input"
\ No newline at end of file
+          description: "Invalid input"
+  /compartments/{id}:
+    get:
+      tags:
+      - "compartment"
+      summary: "For a given compartment name (e.g., Golgi apparatus or
```

Comment:
It works with `golgi_apparatus` but not `Golgi apparatus`, maybe the API can not handle the `id` with spaces?

Reply:
> Another thing I spotted is that the response does not specify which model and version this compartment belongs to. I believe there are some defaults in place that allow the query to succeed, but the query should specify those other parameters in addition to clarifying what the default values are. This seems to apply to the rest of queries.

It seems the syntax for API of compartments is different in `v2` compared to `v1`.  According to https://github.com/MetabolicAtlas/MetabolicAtlas/blob/develop/api/src/endpoints/neo4j.js#L39-L40, one can not specify modelID nor geneID for `/compartments`.

Reply:
The syntax is different indeed, but it still has to take into account the model and other parameters. This is done [inside `fetchWith`](https://github.com/MetabolicAtlas/MetabolicAtlas/blob/develop/api/src/endpoints/neo4j.js#L29). This can be observed in action by looking at the API requests sent from the browser. Here is the [example for `golgi_apparatus`](https://metatlas.csbi.chalmers.se/api/v2/compartments/golgi_apparatus?model=HumanGem&version=1_3_0)

Reply:
Should `modelName` and `version` also be required parameters. Feels like a bit too many input for the users. 
Also, why the model name should be `HumanGem` instead of `Human-GEM` and version be `1_3_0` instead of `1.3.0` in the API?

Reply:
It's on the verbose side, but without these parameters one cannot specify a different model, or a different version. For the near future we don't plan to add multiple versions, so maybe we can skip this parameter. Without a `model`, however, all queries are limited to `HumanGem`.
I think the `-` and `.` were replaced because Neo4j can't handle that character in the label name, but it might be worth double-checking this. I am supportive of finding a way to make the model name and versions consistent.

Reply:
OK. Then I will add `modelName` as a required parameter but ignore `version`. At moment, I will just add example modelName as `HumanGem`, but we can add the task to check whether Eeo4j can handle `-` and `.` in label name or not to the backlog.

Reply:
Hi @mihai-sysbio ,

The parameter `model` is added in the swagger UI. Is there anything else needs to be added? 
If everything works fine. I will merge and close this pull request and go on with the MA-866.

Reply:
The `model` parameter was missing from the last couple of queries. I hope it's okay that I added it.